lib/remap-functions/src/del.rs

```diff
@@ -9,47 +9,58 @@ impl Function for Del {
     }
 
     fn parameters(&self) -> &'static [Parameter] {
-        generate_param_list! {
-            accepts = |_| true,
-            required = false,
-            keywords = [
-                "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16",
-            ],
-        }
+        &[Parameter {
+            keyword: "path",
+            accepts: |_| true,
+            required: true,
+        }]
     }
 
     fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
-        let mut paths = vec![];
-        paths.push(arguments.required_path("1")?);
-
-        for i in 2..=16 {
-            if let Some(path) = arguments.optional_path(&format!("{}", i))? {
-                paths.push(path)
-            }
-        }
+        let path = arguments.required_path("path")?;
 
-        Ok(Box::new(DelFn { paths }))
+        Ok(Box::new(DelFn { path }))
     }
 }
 
 #[derive(Debug, Clone)]
 pub struct DelFn {
-    paths: Vec<Path>,
+    path: Path,
+}
+
+impl DelFn {
+    #[cfg(test)]
+    fn new(path: Path) -> Self {
+        Self { path }
+    }
 }
 
 impl Expression for DelFn {
     fn execute(&self, _: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
-        self.paths
-            .iter()
-            .try_for_each(|path| object.remove(path.as_ref(), false))?;
-
-        Ok(Value::Null)
+        // TODO: we're silencing the result of the `remove` call here, to make
+        // this function infallible.
+        //
+        // This isn't correct though, since, while deleting Vector log
+        // fields is infallible, deleting metric fields is not.
+        //
+        // For example, if you try to delete `.name` in a metric event, the
+        // call returns an error, since this is an infallible field.
+        //
+        // After some debating, we've decided to _silently ignore_ deletions
+        // of infallible fields for now, but we'll circle back to this in
+        // the near future to potentially improve this situation.
+        //
+        // see tracking issue: <TODO>
```

Comment:
Could we open the issue and link it here?

Reply:
Woops, yes, I'll add a link to https://github.com/timberio/vector/issues/5887